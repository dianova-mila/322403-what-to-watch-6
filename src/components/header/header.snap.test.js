import React from "react";
import {Router} from "react-router-dom";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import Header from "./header";
import {createMemoryHistory} from "history";

const authorizationStatus = true;
const userInfo = {avatarUrl: `img/1.png`};

const onUserAvatarClick = jest.fn();
const mockStore = configureStore({});
const history = createMemoryHistory();

describe(`Test header`, () => {
  it(`Header should render correctly when user authorized`, () => {

    const {container} = render(
        <Provider store={mockStore({
          USER: {authorizationStatus, userInfo}
        })}>
          <Router history={history}>
            <Header onUserAvatarClick={onUserAvatarClick}/>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Header should render correctly when user not authorized`, () => {

    const {container} = render(
        <Provider store={mockStore({
          USER: {authorizationStatus: false, userInfo: {}}
        })}>
          <Router history={history}>
            <Header onUserAvatarClick={onUserAvatarClick}/>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
