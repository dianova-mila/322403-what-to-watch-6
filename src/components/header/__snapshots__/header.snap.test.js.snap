// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header Header should render correctly when user authorized 1`] = `
<div>
  <header
    class="page-header movie-card__head"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="user-block"
    >
      <div
        class="user-block__avatar"
      >
        <img
          alt="User avatar"
          height="63"
          src="img/1.png"
          width="63"
        />
      </div>
    </div>
  </header>
</div>
`;

exports[`Test header Header should render correctly when user not authorized 1`] = `
<div>
  <header
    class="page-header movie-card__head"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="user-block"
    >
      <a
        class="user-block__link"
        href="/login"
      >
        Sign in
      </a>
    </div>
  </header>
</div>
`;
